models/src/main/scala/coop/rchain/models/rholang/sorter/ExprSortMatcher.scala

```diff
@@ -259,7 +260,7 @@ private[sorter] object ExprSortMatcher extends Sortable[Expr] {
       case gs: GString => ScoredTerm(e, Node(Score.STRING, Leaf(gs.value))).pure[F]
       case gu: GUri    => ScoredTerm(e, Node(Score.URI, Leaf(gu.value))).pure[F]
       case GByteArray(ba) =>
-        ScoredTerm(e, Node(Score.EBYTEARR, Leaf(ba.toStringUtf8))).pure[F]
+        ScoredTerm(e, Node(Score.EBYTEARR, Leaf(Base16.encode(ba.toByteArray)))).pure[F]
```

Comment:
base16.encode is a hellishly slow way to sort. We should rather use the byte array directly in the leaf, nad make sure that we sort Leaf(byteArray) using the array elements